docs/javascript/compileFaqNode.js

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env node
+
+'use strict';
+
+process.on('uncaughtException', function (err) {
+  console.log('Caught exception: ' + err);
+  process.exit(1);
+});
+
+const faqHtmlTemplate = __dirname + "/" + process.argv[2];
+const pandocOutputDir = __dirname + "/" + process.argv[3];
+const compiledHTMLOutputFile = __dirname + "/" + process.argv[4];
+
+const jsdom = require('jsdom');
+const fs = require('fs');
+
+var $ = null;
+var mjAPI = require("mathjax-node/lib/mj-page.js");
+
+mjAPI.start();
+
+jsdom.env(faqHtmlTemplate, function (err, window) {
+        window.addEventListener("error", function (event) {
+          console.error("script error!!", event.error);
+          process.exit(1);
+        });
+
+        $ = require('jquery')(window);
+
+        function loadReq(selector, file) {
+            return new Promise(function (resolve, reject) {
+                $(selector).load(file, function (response, status, xhr) {
```

Comment:
Figured out the problem. We need to always call `resolve` or `reject`, but if `$(selector)` returns no elements then load doesn't call the callback. So we could do something like:

```javascript
const s = $(selector)
if (s.length == 0) { reject("No elements found for "+selector) }
else { s.load(file, ...) }
```